MetaMask/metamask-extension
# Persist phishing state controller state.

## Explanation
In MV3, when you restart Metamask (or when SW restarts), all the phishing lists are not loaded. This means some phishing sites are not blocked on the first run after a restart.

This PR fixes this issue by persisting PhishingController state.

* Fixes #16551
* See: #16551 for more information and repro steps.
* Blocked By: [PR](https://github.com/MetaMask/controllers/pull/986)

## Manual Testing Steps

1. Start MM in mv3 mode `yarn start:mv3`
2. Load MM extension
3. Load MM UI
4. Go to `https://poccoinss.com/` or `https://polygonteechnology.info`
5. The page should load without the phishing warning the first time you load metamask
6. Loading the page again, you should see the Phishing Warning page
7. Go to extension manager and click on the refresh button to reload metamask
8.  Go to `https://poccoinss.com/` or `https://polygonteechnology.info`
9. The page should load without the phishing warning the first time you load metamask
10. Loading the page again, you should see the Phishing Warning page
11. Checkout this branch and start MM in mv3 mode
12. Repeat the above steps, refresh MM from Extensions Manager and load the phishing pages
13. You should see the phishing warning immediately.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
